pkg/machine/provider/openstack.go

```diff
@@ -115,6 +115,10 @@ func CompleteOpenstackProviderSpec(config *openstack.RawConfig, cluster *kuberma
 				config.TrustDevicePath.Value = ptr.To(false)
 			}
 		}
+
+		if config.ConfigDrive.Value == nil && datacenter.EnableConfigDrive != nil {
+			config.ConfigDrive.Value = datacenter.EnableConfigDrive
+		}
```

Comment:
This change will only affect the [initial machine deployments](https://github.com/kubermatic/kubermatic/blob/main/pkg/controller/seed-controller-manager/initial-machinedeployment-controller/machinedeployment.go#L118-L122) created during cluster creation. All machine deployments created after cluster creation from UI or otherwise will not be affected by this change. Hence, you'll get initial machine deployments with the config drive enabled, and other machine deployments will have the config drive disabled.

Can you please test and confirm this behavior?

Reply:
That's right correct, but machines that are created in UI will be addressed in the dashboard repo. However, this logic regardless is correct.

Reply:
I'm sorry, but as planned, there won't be dashboard implementation in this release, as you already know. So, in this case, the customer who requested this feature will already be affected by this issue.

Reply:
Machines that are not created via UI, the user directly creates `MachineDeployment` in the user cluster, are also not governed by this chunk of code. So, they will also not have the config drive enabled, even if it's enabled at the data center level.

Reply:
Not entirely sure what do you mean tbh, however will take care of that as well 😉

Reply:
I'll just tag @xrstf and maybe he'll be able to explain this to you in a better way.

Reply:
As already mentioned, for MachineDeployment that are created in the dashboard there must be a defaulting behavior as well. AFAIR, there is no way that we can control the creation of the machine deployment using the api server directly, except sending values to machine controller binary or some hardcoded values that read from a config in the cluster namespace. It has been like this since the start and I don't remember that this has changed. Which means any machine deployment that will be created using the api directly, won't have any KKP validation. If that changed somewhere please point me to that functionality,

Reply:
Both of you are right in everything you said.

The only disagreement/issue I can see here is that for KKP2.26, the UI will not support this new feature. As Moath pointed out, KKP does not do any defaulting (MC has its own webhook, but of course no knowledge of KKP's interna), so there is no way for KKP to inject the necessary flags into the MD as it travels from the kubermatic-api to the usercluster's kube-apiserver.

So there is, for a user sitting in front of a KKP 2.26 system, a difference in how Machines behave that were created during the cluster wizard and those created later. I think this friction is what Waleed wanted to highlight. There is also a chance that you lose the new field from the MD's spec if you edit the MD in the dashboard (not sure though how exactly we patch the MD).

A way to auto-inject this into every Machine would be to add new flags to the machine-controller, which could be set by KKP. But normally we don't do this.